lib/surface/compiler/eex_engine.ex

```diff
@@ -224,7 +227,7 @@ defmodule Surface.Compiler.EExEngine do
         unquote(module),
         Surface.build_assigns(
           unquote(assigns_expr),
-          unquote(props_expr),
+          Keyword.merge(unquote(props_expr), unquote(dynamic_props_expr)),
```

Comment:
I wonder if we could wrap this logic in a separate function or maybe even move it to `Surface.build_assign` altogether in order to have a cleaner `live_component` call. Currently, the generated code gets too verbose with the two nested `for`:

```jsx
<DivWithProps content="static content" :props={{ class: "text-xs", hidden: false, content: "dynamic props content" }} :debug/>
```
Generates:

```elixir
live_component(
  @socket,
  Surface.DirectivesTest.DivWithProps,
  Surface.build_assigns(
    @__surface__[:context] || [],
    Keyword.merge(
      [class: nil, hidden: nil, content: "static content"],
      for(
        {name, {type, value}} <-
          for(
            {name, value} <-
              Surface.TypeHandler.expr_to_value!(
                :map,
                ":props",
                [],
                [class: "text-xs", hidden: false, content: "dynamic props content"],
                Surface.DirectivesTest.DivWithProps,
                " class: \"text-xs\", hidden: false, content: \"dynamic props content\" "
              ) || [],
            not Enum.member?([:content], name)
          ) do
            {name,
             Surface.TypeHandler.runtime_prop_value!(Surface.DirectivesTest.DivWithProps, name, value, "DivWithProps")}
          end
      ) do
        {name, value}
      end
    ),
    [],
    [__default__: %{size: 0}],
    Surface.DirectivesTest.DivWithProps
  ),
  []
)
```

Is it feasible?

Reply:
I pushed up a change that moved both handling the prop defaults and the dynamic props into `Surface.build_assigns` which generates a much cleaner `live_component` call:

```elixir
live_component(
  @socket,
  Surface.DirectivesTest.DivWithProps,
  Surface.build_assigns(
    @__surface__[:context] || [],
    [content: "static content"],
    Surface.TypeHandler.expr_to_value!(
      :map,
      ":props",
      [],
      [class: "text-xs", hidden: false, content: "dynamic props content"],
      Surface.DirectivesTest.DivWithProps,
      " class: \"text-xs\", hidden: false, content: \"dynamic props content\" "
    ),
    [],
    [__default__: %{size: 0}],
    Surface.DirectivesTest.DivWithProps,
    "DivWithProps"
  ),
  []
)
```

It also cleaned up/simplified the directive code a bit :-)